### PR TITLE
chore: upgrade Jest to 23

### PR DESCRIPTION
**Pull request by thymikee**
**Summary**

Bumps Jest to 22 (20 is pretty old now). Had to introduce some minor changes, as I've upgraded related deps by the way (e.g. `puppeteer`).

This PR makes it possible to test under Node 10 (it hanged on react-native test for reasons, no time to investigate)

### CODE REVIEW

File: src/test/warnTooManyClasses.test.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ import styleSheet from '../models/StyleSheet'
 let styled
 
 describe('warn too many classes', () => {
-  const nativeWarn = console.warn
```
Comment (probablyup) [inline src/test/warnTooManyClasses.test.js]: Hmm why rename this? It was correctly described before, meaning native console.warn vs a mocked console.

File: test-utils/setupTestFramework.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  ;(console: any).error = message => {
+    if (!message.includes('Error: Could not parse CSS stylesheet')) {
+      consoleError(message)
+    }
+  }
+})
+
+afterEach(() => {
+  // eslint-disable-next-line flowtype-errors/show-errors
+  ;(console: any).error = consoleError
```
Comment (probablyup) [inline test-utils/setupTestFramework.js]: `;(console: any).error` what is this magic?

File: src/test/warnTooManyClasses.test.js (hunk 1 above)
Comment (thymikee) [reply src/test/warnTooManyClasses.test.js]: Can revert this, it's a leftover actually.

File: test-utils/setupTestFramework.js (hunk 2 above)
Comment (thymikee) [reply test-utils/setupTestFramework.js]: Ask Prettier :D

Comment (kitten) [reply test-utils/setupTestFramework.js]: ha! It'd make sense if the previous line required a semicolon to be put there but better safe than sorry ?! 😆

**Comment by thymikee** [comment]
Seems it's not complaining :)

**Comment by probablyup** [comment]
Thanks!

**Pull request merged**